script/draft-release/version.ts

```diff
@@ -73,7 +73,33 @@ export function getNextVersionNumber(
         const firstBeta = `${nextVersion}-beta1`
         return firstBeta
       }
+    case 'test':
+      console.warn(
+        `WARNING: Since we do not tag test versions, the version provided will always be the first of test series respective of latest prod or beta.
+         You may need to adjust the -testX according to how many test releases already exist.`
+      )
 
+      if (isTestTag(semanticVersion)) {
+        /*Since we do not tag test versions, it "should be" impossible to reach
+        this first if, but added just for the case of future proofing the ideal
+        path. */
```

Comment:
Maybe we should tag them? In the end the code is public, there is a track record of them from Central. I don't think it's that bad, and it's always kind of annoying if someone else has been pushing test builds you have to take a look at the latest one and +1 whatever build number that has.

Another option is to maybe allow specifying a specific "test build number" as a parameter for test builds.

Reply:
I went ahead and investigated what else needs to be done to fully support the test channel. I have added the changes here, but for full support we need to create the tags which is done during deployment. I have added the necessary change to [here](https://github.com/github/central/pull/575).